docs/guides.md

```diff
@@ -523,3 +523,307 @@ Now we enable autostart at the virtual machine level.
+
+Something to note when creating the file is to allow extra space between command line parameters.  That is why most of the lines have extra space before and after parameters within the quotes.
+
+### Building and Testing the New intrd File
+
+Now that we have all of this extra configuration, we need to get the system to include it for use at boot.  To do that we use the dracut command to create a new initrd file like this:
+
+```
+dracut --force -M /boot/initrd-$(uname -r).img $(uname -r)
+```
+
+This creates a new initrd file with the correct name matching the name of the Linux kernel and prints a list of modules included in the initrd file.  Printing the list isn't necessary but is handy to see that dracut is making progress as it runs.
+
+When the system returns to the command line, it's time to test.  We'll reboot the system from the console or from within Xen Orchestra or XCP-ng Center.  If all goes well, the system should boot normally and correctly find and mount all 5 drives into the two RAID arrays.  The easiest way to tell that is looking at the /proc/mdstat file.
+
+```
+[14:36 XCP-ng ~]# cat /proc/mdstat
+Personalities : [raid1] [raid6] [raid5] [raid4]
+md127 : active raid1 sda[0] sdb[1]
+      976762432 blocks super 1.0 [2/2] [UU]
+      bitmap: 1/8 pages [4KB], 65536KB chunk
+
+md0 : active raid5 sdc[0] sde[2] sdd[1]
+      1953260544 blocks super 1.2 level 5, 512k chunk, algorithm 2 [3/3] [UUU]
+      bitmap: 3/8 pages [12KB], 65536KB chunk
+
+unused devices: <none>
+```
+
+We can see that both arrays are active and healthy with all drives accounted for.  Examining the storage repositories using Xen Orchestra, XCP-ng, or xe commands shows that both the Local storage and RAID storage repositories are available.
+
+### Troubleshooting
+
+The most common problems in this process stem from one of a few things.
+
+One common cause of problems is using the wrong type of drives.  Just like when using drives for installation of XCP-ng, it is important to use drives that either have or emulate 512 byte disk sectors.  Drives that use 4K byte disk blocks will not work unless they are 512e disks which emulate having 512 byte sectors.  It is generally not a good idea to mix types of drives such as one 512n (native 512 byte sectors) and two 512e drives but should be possible to do in an emergency.
+
+The second is that the drives were not empty before including them into the system.  If there are any traces of RAID configuration or file systems on the drives, we could have problems with interference between those and the new configurations we're creating on the drives when creating the RAID array or the EXT filesystem (or LVM if you use that for the storage array).
+
+The way to avoid this problem is to make sure the drives are thoroughly wiped before starting the process.  This can be done from the command line with the dd command like this:
```

Comment:
```suggestion
The way to avoid this problem is to make sure the drives are thoroughly wiped before starting the process.  This can be done from the command line with the `dd` command like this:
```